tests/template_tests/test_loaders.py

```diff
@@ -126,18 +130,60 @@ def test_templatedir_caching(self):
         self.assertNotEqual(t1.render(Context({})), t2.render(Context({})))
 
     def test_missing_template_is_cached(self):
-        "#19949 -- Check that the missing template is cached."
-        # Check that 'missing.html' isn't already in cache before 'missing.html' is loaded
+        """
+        #19949 -- TemplateDoesNotExist exceptions should be cached.
+        """
+        # Check that 'missing.html' isn't already in the cache
         with self.assertRaises(KeyError):
             self.loader.template_cache["missing.html"]
         # Try to load it, it should fail
         with self.assertRaises(TemplateDoesNotExist):
             self.loader.load_template("missing.html")
-        # Verify that the fact that the missing template, which hasn't been found, has actually
-        # been cached:
+        # Check that the exception has been cached
         cached_miss = self.loader.template_cache["missing.html"]
-        self.assertEqual(cached_miss, TemplateDoesNotExist,
-                         "Cached template loader doesn't cache file lookup misses. It should.")
+        self.assertEqual(
+            cached_miss, TemplateDoesNotExist,
+            "Cached loader failed to cache the TemplateDoesNotExist exception",
+        )
+
+
+class GetTemplateTests(SimpleTestCase):
```

Comment:
If you were testing the `get_template` function then the tests would go in the `template_loader` test app. Can you give this class another name (or move it)?

By the way, the tests would be much cleaner if they created a suitable `Engine` in `setUp` and ran the tests on that engine instead of the global one, like `template_tests.utils.setup` does.

I don't want to send you down a rabbit hole when you were just doing some light cleanup :-) but that's something to keep in mind if you add new tests.